litex_setup.py

```diff
@@ -29,18 +31,69 @@
     ("litespi",      ("https://github.com/litex-hub/",     False, True)),
 
     # LiteX boards support
-    ("litex-boards",   ("https://github.com/litex-hub/", False, True)),
+    ("litex-boards", ("https://github.com/litex-hub/",     False, True)),
 ]
 repos = OrderedDict(repos)
 
+
+def sifive_riscv_download():
+    base_url = "https://static.dev.sifive.com/dev-tools/"
+    base_file = "riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-"
+
+    is_windows = (
+        sys.platform.startswith('win') or sys.platform.startswith('cygwin'))
+    if is_windows:
+        end_file = 'w64-mingw32.zip'
+    elif sys.platform.startswith('linux'):
+        end_file = 'linux-ubuntu14.tar.gz'
+    elif sys.platform.startswith('darwin'):
+        end_file = 'apple-darwin.tar.gz'
+    else:
+        raise NotImplementedError(sys.platform)
+    fn = base_file + end_file
+
+    if not os.path.exists(fn):
+        url = base_url+fn
+        print("Downloading", url, "to", fn)
+        urllib.request.urlretrieve(url, fn)
+    else:
+        print("Using existing file", fn)
+
+    print("Extracting", fn)
+    shutil.unpack_archive(fn)
+
+    if "--user" in sys.argv[1:] and not is_windows:
```

Comment:
This code symlinks the risc-v compiler into your `~/.local/bin` directory. Now that I have added this, I think it could be a mistake to do this and it does add a lot of extra code here.